src/data-table/types.ts

```diff
@@ -0,0 +1,83 @@
+import { CSSProperties, ReactNode } from "react";
+import { ErrorDisplayAttributes } from "../error-display/types";
+export interface DataTableProps {
+    id?: string | undefined;
+    "data-testid"?: string | undefined;
+    headersConfig?: HeadersConfigProps | undefined;
+    headers: HeaderProps[];
+    rowsConfig?: RowsConfigProps | undefined;
+    rows?: RowProps[] | undefined;
+    /** css class to put on table **/
+    className?: string | undefined;
+    selectionConfig?: SelectionConfigProps | undefined;
+    /** ids of all selected rows **/
+    selection?: string[] | undefined;
+    /** columns that want to show a sort indicator **/
+    sortIndicators?: SortIndicatorsProps | undefined;
+    // actionsConfig?: ActionsConfigProps | undefined;
+    alternatingRows?: boolean | undefined;
+    customEmptyView?: () => ReactNode | string | undefined;
+    loadState: LoadingType | undefined;
+    emptyView?: ErrorDisplayAttributes | undefined;
+}
+
+export type LoadingType = "loading" | "success" | "error";
```

Comment:
if there's no error state specification, we can leave it out for now to avoid unintended usage